src/language-html/print/angular-control-flow-block.js

```diff
@@ -64,4 +78,81 @@ function printAngularControlFlowBlockParameters(path, options, print) {
   ];
 }
 
+function lineStart(node) {
+  return node.sourceSpan.start.line;
+}
+
+function lineEnd(node) {
+  return node.sourceSpan.end.line;
+}
+
+function isWhitespaceSensitiveNodeInBlock(node) {
+  return node.type === "text" || node.type === "interpolation";
+}
+
+function allSiblingsSpaceSensitive(block, child) {
+  for (const ch of block.children) {
+    if (ch === child) {
+      continue;
+    }
+    if (!isWhitespaceSensitiveNodeInBlock(ch)) {
+      return false;
+    }
+  }
+  return true;
+}
+
+/**
+ * Returns whether to print a hardline inside the brackets of a control flow block.
+ *
+ * The inside of Angular control flow blocks is space-sensitive,
+ * meaning the following two pieces of code produce different outputs:
+ *
+ * 1:
+ *   @if (true) {foo}
+ *
+ * 2:
+ *   @if (true) {
+ *     foo
+ *   }
+ *
+ * So, when the --html-whitespace-sensitivity option is not set to `ignore`,
+ * we should control line breaks to ensure consistent rendering.
+ * For more details, please read https://github.com/prettier/prettier/issues/16577.
+ *
+ * @param {any} block
+ * @param {any} options
+ * @param {"open" | "close"} kind
+ * @returns {boolean}
+ */
+function shouldBreakLineWithinBrackets(block, options, kind) {
+  if (options.htmlWhitespaceSensitivity === "ignore") {
```

Comment:
I don't think this is the correct solution, it should check if the parent element is whitespace sensitive.